### PR TITLE
Reworking hot reload config tests

### PR DESCRIPTION
**Pull request by corey-nm**
The previous way of spinning up a separate server process had a couple problems:
1. it was flaky
2. it was slow
3. it didn't really work well on github actions

This PR removes that test and replaces it with:
1. Tests that the hot_config_reload flag of start_server properly calls start_config_watcher
2. Tests that the cli commands pass the proper flag to start_server

All of this is done with mocking and click CLIRunner, so its really fast

### CODE REVIEW

File: tests/server/test_hot_reloading.py [hunk 1]
```diff
@@ -166,82 +164,48 @@ def test_file_monitoring(delete_mock, post_mock, tmp_path: Path):
...
-    assert resp.status_code == 404
+    start_server(cfg_path, hot_reload_config=False)
+    run_patch.assert_called_once()
+    watcher_patch.assert_not_called()
+
+    run_patch.reset_mock()
+    watcher_patch.reset_mock()
+    start_server(cfg_path, hot_reload_config=True)
+    run_patch.assert_called_once()
+    watcher_patch.assert_called_once()
```
Comment (bfineran) [inline tests/server/test_hot_reloading.py:185]: let's add in some inline comments for what's being tested

**Review by bfineran — COMMENTED** [review]
quick comment

File: tests/server/test_hot_reloading.py (hunk 1 above)
Comment (corey-nm) [reply tests/server/test_hot_reloading.py]: Good point, added comments and improved skimmability of them

**Review by rahul-tuli — APPROVED** [review]
To the moon!

**Review by KSGulin — APPROVED** [review]

**Pull request merged**